packages/asset-swapper/src/network/samplers/curve.ts

```diff
@@ -0,0 +1,709 @@
+    [FIREBIRDONESWAP_POLYGON_POOLS.oneswap]: {
+        exchangeFunctionSelector: CurveFunctionSelectors.swap,
+        sellQuoteFunctionSelector: CurveFunctionSelectors.calculateSwap,
+        buyQuoteFunctionSelector: CurveFunctionSelectors.None,
+        poolAddress: FIREBIRDONESWAP_POLYGON_POOLS.oneswap,
+        tokens: [POLYGON_TOKENS.DAI, POLYGON_TOKENS.USDC, POLYGON_TOKENS.USDT],
+        metaTokens: undefined,
+        gasSchedule: 100e3,
+    },
+};
+const CURVELIKE_INFOS_BY_CHAIN_ID = valueByChainId({
+        [ChainId.Mainnet]: {
+            [ERC20BridgeSource.Curve]: CURVE_MAINNET_INFOS,
+            [ERC20BridgeSource.Swerve]: SWERVE_MAINNET_INFOS,
+            [ERC20BridgeSource.SnowSwap]: SNOWSWAP_MAINNET_INFOS,
+            [ERC20BridgeSource.Saddle]: SADDLE_MAINNET_INFOS,
+            [ERC20BridgeSource.XSigma]: XSIGMA_MAINNET_INFOS,
+            [ERC20BridgeSource.CurveV2]: CURVE_V2_MAINNET_INFOS,
+        },
+        [ChainId.BSC]: {
+            [ERC20BridgeSource.Nerve]: NERVE_BSC_INFOS,
+            [ERC20BridgeSource.Belt]: BELT_BSC_INFOS,
+            [ERC20BridgeSource.Ellipsis]: ELLIPSIS_BSC_INFOS,
+            [ERC20BridgeSource.FirebirdOneSwap]: FIREBIRDONESWAP_BSC_INFOS,
+        },
+        [ChainId.Polygon]: {
+            [ERC20BridgeSource.Curve]: CURVE_POLYGON_INFOS,
+            [ERC20BridgeSource.CurveV2]: CURVE_V2_POLYGON_INFOS,
+            [ERC20BridgeSource.FirebirdOneSwap]: FIREBIRDONESWAP_POLYGON_INFOS,
+        },
+    },
+    {},
+) as any as { [k in ChainId]: { [k in ERC20BridgeSource]: { [name: string]: CurveInfo }} };
+
+type SellContract = ERC20BridgeSamplerContract;
+type BuyContract = ERC20BridgeSamplerContract;
+type SellContractSellFunction = SellContract['sampleSellsFromCurve'];
+type BuyContractBuyFunction = BuyContract['sampleBuysFromCurve'];
+type SellContractSellFunctionParameters = Parameters<SellContractSellFunction>;
+type BuyContractBuyFunctionParameters = Parameters<BuyContractBuyFunction>;
+type SellContractSellFunctionReturn = UnwrapContractFunctionReturnType<ReturnType<SellContractSellFunction>>;
+type BuyContractBuyFunctionReturn = UnwrapContractFunctionReturnType<ReturnType<BuyContractBuyFunction>>;
+type SamplerSellEthCall = SamplerEthCall<CurveFillData,SellContractSellFunctionParameters,SellContractSellFunctionReturn>;
+type SamplerBuyEthCall = SamplerEthCall<CurveFillData,BuyContractBuyFunctionParameters,BuyContractBuyFunctionReturn>;
+
+export class CurveSampler extends
+    OnChainSourceSampler<
+        SellContract,
+        BuyContract,
+        SellContractSellFunction,
+        BuyContractBuyFunction,
+        CurveFillData
+    >
+{
+    public static async createAsync(
+        chain: Chain,
+        fork: ERC20BridgeSource,
+    ): Promise<CurveSampler> {
+        const curveInfos = CURVELIKE_INFOS_BY_CHAIN_ID[chain.chainId];
+        if (!curveInfos) {
+            throw new Error(`No curve configs for chain ${chain.chainId}`);
```

Comment:
Is it bad do have sitting around with empty CurveInfos to act upon?

Reply:
Not bad, just pointless. I'm pretty inconsistent on checking for valid configs in these samplers.